### PR TITLE
Adding a stage to the build in which the full corpus is tested

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Build took 10 minutes for the generation, which can possibly be improved (with an even more powerful machine). However the validation process stopped immediately for a problematic import? @lsh-0

**Comment by lsh-0** [comment]
is this your error?
```
17:19:49 [5-R6OZRHTKGH362ZAT2N6NSPZDWRXAZESSG4UZ3ZKOBF7NUBCSSCWQ] Running shell script
17:19:49 + ./validate-json.sh
17:19:49 Traceback (most recent call last):
17:19:49   File "src/validate.py", line 2, in <module>
17:19:49     import conf
17:19:49   File "/ext/jenkins-libraries-runner/workspace/5-R6OZRHTKGH362ZAT2N6NSPZDWRXAZESSG4UZ3ZKOBF7NUBCSSCWQ/src/conf.py", line 3, in <module>
17:19:49     from pythonjsonlogger import jsonlogger
17:19:49 ImportError: No module named pythonjsonlogger
```
looking at `validate-json.sh` the call to `install.sh` that activates the venv is commented out. I don't know how it was able to make it as far as it did, validating all those articles, without a virtualenv.

**Comment by lsh-0** [comment]
nuts, sorry. I changed this to `validate-all-json.sh` so I could have scripts that could validate a single article, scrape a single article, scrape all, validate all, scrape random, etc

**Comment by lsh-0** [comment]
fixed it up

**Comment by lsh-0** [comment]
```
12:18:25 + ./download-elife-xml.sh
12:18:25 Cloning into 'article-xml'...
12:18:25 fatal: '/elife-article-xml' does not appear to be a git repository
12:18:25 fatal: Could not read from remote repository.
12:18:25 
12:18:25 Please make sure you have the correct access rights
12:18:25 and the repository exists.
```

**Comment by gnott** [comment]
I couldn't reproduce an error scraping 00240 yesterday, but today it was not scraping as well as a dozen or so other articles. I am fixing elifetools so these will scrape again. That may not have been the primary issue here, but that's what I noticed in the console log.

**Comment by lsh-0** [comment]
there is a ticket for me to tweak the ingestion and validation so we get Alfred failing builds if any of the articles fail to generate or validate against the corpus

**Comment by giorgiosironi** [comment]
To be precise, since many are failing now, the check would be "if more than N articles from the corpuse fail to validate" where N is the baseline.

**Comment by giorgiosironi** [comment]
We may be losing 2 articles somewhere in the generation:
```
17:00:15 + ./generate-statistics.sh corpus-generation.log
17:00:15 Input: 4875 XML articles
17:00:15 Generated green,red: 4873,0
```
The `0` should be `2`, have to check why it's not.

```
17:04:28 + ./validate-statistics.sh corpus-validation.log
17:04:28 Input: 4875 JSON articles
17:04:28 Validated green,red: 4783,91
```
The sum here makes it for 4874 too, which means we don't know yet the status of 1.

**Comment by gnott** [comment]
Any chance the 1 that doesn't parse is the ``.placeholder`` file in ``article-xml/articles`` directory?

**Comment by giorgiosironi** [comment]
Latest results:
```
17:44:34 Input: 4875 XML articles
17:44:34 Generated green,red,total: 4874,0,4874
17:48:41 Input: 4875 JSON articles
17:48:41 Validated green,red: 4783,91,4874
```
I'll set the target to 91 and check the discrepancy between inputs and outputs.

**Comment by giorgiosironi** [comment]
The one that we don't have in the output is `article-xml/articles/elife-08658-v1.xml`. Its .json is probably correctly generated because we see 4875 JSON articles, but it does not appear in the log.

**Comment by giorgiosironi** [comment]
The one that is missing from the validation log instead is ` elife-07861-v2`:
```
$ grep -r 07861 ~/corpus-validation.log 
INFO - validated 07861
INFO - 2016-11-21 17:46:45,420 - validated 07861
elife-07861-v1.xml.json => success
INFO - validated 07861
INFO - 2016-11-21 17:46:46,084 - validated 07861
```

**Comment by giorgiosironi** [comment]
Logging a lot more to find out what's happening.

File: src/validate.py [hunk 1]
```diff
@@ -11,12 +11,6 @@
 _handler.setFormatter(conf._formatter)
 LOG.addHandler(_handler)
 
-# output to screen
```
Comment (giorgiosironi) [inline src/validate.py]: this duplicates `validate_article_json.py`

**Comment by giorgiosironi** [comment]
After changing the log creation, success for the first time in the last 24 hours. Ready for review.

**Pull request merged**